metricbeat/module/kubernetes/state_container/data.go

```diff
@@ -0,0 +1,89 @@
+package state_container
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/kubernetes/util"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+const (
+	// Nanocores conversion 10^9
+	nanocores = 1000000000
+)
+
+func eventMapping(families []*dto.MetricFamily) ([]common.MapStr, error) {
```

Comment:
That was my initial intention, but I have 2 issues with that:

- Prometheus module mangles original data on it's way, and it's not really aware of the format of these metrics, it does something generic. So I need to go over the resulting MapStr again
- If we do a change in prometheus module it may break this, having that I'm not using its format

Something I thought about was to move all prometheus code to a common helper (like HTTP), WDYT about that? At the end it looked like overkill but I'm ok with doing it